presidio-analyzer/Dockerfile.transformers

```diff
@@ -0,0 +1,22 @@
+FROM python:3.9-slim
+
+ARG NAME
+ARG NLP_CONF_FILE=conf/transformers.yaml
+ENV PIPENV_VENV_IN_PROJECT=1
+ENV PIP_NO_CACHE_DIR=1
+WORKDIR /usr/bin/${NAME}
+
+COPY ./Pipfile* /usr/bin/${NAME}/
+RUN pip install pipenv \
+  && pipenv sync
+RUN pipenv install torch transformers huggingface_hub --skip-lock
```

Comment:
This is the main change between the transformers version and the previous one. Any ideas on how we can have those under one Dockerfile?

Reply:
Do you mean a single file that can produce 2 images?

Reply:
So a build arg that would conditionally run a line? Some like this?
https://github.com/microsoft/AzureTRE/blob/main/.devcontainer/Dockerfile#L45

Reply:
Yes. Thanks for the reference

Reply:
Personally I think that is not a best practice, since the produced docker image is not consistent.

Reply:
@balteravishay then any build arg will produce an inconsistent image...

Reply:
Another, more elaborate way would be multistage with "conditional" stages.

Reply:
I'm not sure multistage with "conditional" stages is supported without docker buildkit.

Reply:
And that's not something you want to use?

Reply:
At and case it would be an arg of some sort that would change how the image is built. So at the base of it, an input can produce different images but I think it's pretty common scenario.

Reply:
What's the verdict? Having one Dockerfile that could produce images? or having two `Dockerfile` and `Dockerfile.transformers` files?

Reply:
Its a good practice to not use conditions in the dockerfile (use one dockerfile) since you are not using the layer caching option. 
What you can do have a base dockerfile and extend it to create flavours of it

Reply:
talked with @tamirkamara and we can proceed with having a single dockerfile, and using buildkit.

Reply:
Thanks. That's quite new to me. WDYT about the existing solution (two dockerfiles) as a first step, and introducing buildkit as a second? Not optimal, only temporary.